components/table/Table.tsx

```diff
@@ -1028,7 +1025,7 @@ export default class Table<T> extends React.Component<TableProps<T>, TableState<
     if (
       data.length > pageSize ||
       pageSize === Number.MAX_VALUE ||
-      current * pageSize > data.length
+      (typeof pagination.current === 'undefined' && current * pageSize > data.length)
```

Comment:
directly `pagination.current === undefined`

Reply:
Oh, I'm used to writing like this to avoid `var undefined = 0`  from other scripts.